### PR TITLE
Remove per-operation `Open()`/`Close()` calls to BoltDB

### PR DESCRIPTION
**Pull request by js-ojus**
* `Context` now maintains a DB handle that is kept alive

### CODE REVIEW

File: cmd/drive/main.go [hunk 1]
```diff
@@ -410,6 +410,8 @@ func (cmd *indexCmd) Run(args []string) {
 	byId := *cmd.byId
 	byMatches := *cmd.matches
 	sources, context, path := preprocessArgsByToggle(args, byMatches || byId)
+	context.OpenDB()
```
Comment (odeke-em) [inline cmd/drive/main.go]: We'll need to handle at least OpenDB() errors, please set it to
```go
exitWithError(context.OpenDB())
```

File: cmd/drive/main.go [hunk 2]
```diff
@@ -493,6 +495,8 @@ func (cmd *pullCmd) Flags(fs *flag.FlagSet) *flag.FlagSet {
 
 func (cmd *pullCmd) Run(args []string) {
 	sources, context, path := preprocessArgsByToggle(args, (*cmd.byId || *cmd.matches))
+	context.OpenDB()
```
Comment (odeke-em) [inline cmd/drive/main.go]: Same thing here for OpenDB() and everywhere we have OpenDB()

**Comment by odeke-em** [comment]
I've added a few comments.
Also please do this for indexCmd
```diff
--git a/cmd/drive/main.go b/cmd/drive/main.go
index b417f20..f43d837 100644
--- a/cmd/drive/main.go
+++ b/cmd/drive/main.go
@@ -411,6 +411,9 @@ func (cmd *indexCmd) Run(args []string) {
 	byMatches := *cmd.matches
 	sources, context, path := preprocessArgsByToggle(args, byMatches || byId)
 
+	exitWithError(context.OpenDB())
+	defer context.CloseDB()
+
 	options := &drive.Options{
 		Sources:           sources,
 		Hidden:            *cmd.hidden,
```

**Comment by js-ojus** [comment]
I have made the suggested changes @odeke-em!  Please take a look at it again.

File: config/config.go [hunk 3]
```diff
@@ -311,6 +268,17 @@ func (c *Context) DeInitialize(prompter func(...interface{}) bool, returnOnAnyEr
...
 
+func (c *Context) OpenDB() (err error) {
+	dbPath := DbSuffixedPath(c.AbsPathOf(""))
+	db, err := bolt.Open(dbPath, O_RWForAll, nil)
+	c.DB = db
+	return
+}
+
+func (c *Context) CloseDB() error {
+	return c.DB.Close()
```
Comment (odeke-em) [inline config/config.go]:
```go
if c.DB == nil {
    return errors.New("nil dereference of db")
}
return c.DB.Close()
```

**Comment by odeke-em** [comment]
Thanks @js-ojus, just one more and then we are good to go.

**Comment by odeke-em** [comment]
Once that is in, please help me squash the commits into one and then I'll merge it in.
Thank you again.

**Comment by js-ojus** [comment]
Hello @odeke-em!  I have squashed the commits into a single one, which is now reflected into the pull request.

**Comment by odeke-em** [comment]
LGTM! Thank you very much for diagnosing, and working on this.

**Pull request merged**